Distributions.cpp

```diff
@@ -0,0 +1,168 @@
+ @details   This file contains functions to be used in a Monte Carlo simulation
+            to sample the various distributions arising in reactor physics.
+ @author    Luke Eure
+ @date      January 8, 2016
+*/
+
+#include "Distributions.h"
+
+/*
+  @brief    Returns a uniform random number between 0 and 1
+  @return   a double in the range (0, 1)
+*/
+double urand() {
+    return (double) rand() / (double) RAND_MAX;
+}
+
+/*
+  @brief    Function that randomly samples an azimuthal angle                   
+  @details  An azimuthal angle is uniformally sampled in [0, 2 pi]              
+  @return   A randomly sampled angle in [0, 2pi] 
+*/
+double sampleAzimuthalAngle() {
+    return 2 * M_PI * urand();
+}
+
+/*
+ @brief     Function that randomly samples a polar angle
+ @details   The cosine of the polar angle is uniformally sampled in [-1, 1]
+            and then transformed to the polar angle with an inverse cosine
+ @return    A randomly sampled polar angle in [0, pi]
+*/
+double samplePolarAngle() {
+    double cos_theta = 2 * urand() - 1.0;
+    return acos(cos_theta);
+}
+
+/*
+ @brief     Function that samples the distance of travel based on a total
+            cross-section.
+ @details   A distance of travel is sampled in [0, infinity) assuming an
+            exponential distibution of the form sigma_t * exp(-sigma_t * d).
+ @param     mat a Material object that contains information about the material
+ @param     group an int denoting the energy group of the neutron
+ @return    A randomly sampled distance in [0, infinity)
+*/
+double sampleDistance(Material* mat, int group) {
+    return -log(urand()) / mat->getSigmaT(group);
+}
+
+/*
+ @brief     Function that samples the interaction type (0 = scattering,
+            1 = absorption)
+ @details   Based on cross-sections, the interaction is sampled as scattering
+            (0) or absorption (1).
+ @param     mat a Material object that contains information
+            about the material
+ @param     group an int denoting the energy group of the neutron 
+ @return    An interaction type (0 = scattering, 1 = absorption)
+*/
+int sampleInteraction(Material* mat, int group) {
+    return (int) (urand() < mat->getSigmaA(group)
+                / mat->getSigmaT(group));
+}
+
+/*
+ @brief     Function that samples a random location within a bounding box.
+ @details   A point is randomly and uniformally sampled in the bounding box 
+            provided in the input.
+ @param     bounds a Boundaries object containing the limits of the
+            bounding box
+ @return    _dist_location a vector that contains the coordinates of a point
+*/
+std::vector <double> sampleLocation(Boundaries bounds) {
+    std::vector <double> _dist_location;
+    double width;
+    double coord;
+    for (int axis=0; axis<3; ++axis) {
+        width = bounds.getSurfaceCoord(axis, 1)
+            - bounds.getSurfaceCoord(axis, 0);
+        coord = bounds.getSurfaceCoord(axis, 0) + width * urand();
```

Comment:
Stylistically, it would be better to do:
```
double width = bounds.getSurfaceCoord(axis, 1)
                        - bounds.getSurfaceCoord(axis, 0);
double coord = bounds.getSurfaceCoord(axis, 0) + width * urand();
```